tedana/metrics/dependence.py

```diff
@@ -0,0 +1,527 @@
+"""
+Fit models.
+"""
+import logging
+
+import numpy as np
+from scipy import stats
+
+from tedana import io, utils
+from tedana.stats import computefeats2, get_coeffs, t_to_z
+
+
+LGR = logging.getLogger(__name__)
+RepLGR = logging.getLogger('REPORT')
+RefLGR = logging.getLogger('REFERENCES')
+
+
+def calculate_weights(data_optcom, mixing):
+    """
+    Calculate standardized parameter estimates between data and mixing matrix.
+
+    Parameters
+    ----------
+    data_optcom : (S x T) array_like
+        Optimally combined data
+    mixing : (T x C) array_like
+        Mixing matrix
+
+    Returns
+    -------
+    weights : (S x C) array_like
+        Standardized parameter estimates for optimally combined data against
+        the mixing matrix.
+    """
+    assert data_optcom.shape[1] == mixing.shape[0]
+    mixing_z = stats.zscore(mixing, axis=0)
+    # compute un-normalized weight dataset (features)
+    weights = computefeats2(data_optcom, mixing_z, normalize=False)
+    return weights
+
+
+def calculate_betas(data_optcom, mixing):
+    """
+    Calculate unstandardized parameter estimates between data and mixing
+    matrix.
+
+    Parameters
+    ----------
+    data_optcom : (S x T) array_like
+        Optimally combined data
+    mixing : (T x C) array_like
+        Mixing matrix
+
+    Returns
+    -------
+    betas : (S x C) array_like
+        Unstandardized parameter estimates
+    """
+    assert data_optcom.shape[1] == mixing.shape[0]
+    # demean optimal combination
+    data_optcom_dm = data_optcom - data_optcom.mean(axis=-1, keepdims=True)
+    # compute PSC dataset - shouldn't have to refit data
+    betas = get_coeffs(data_optcom_dm, mixing)
+    return betas
+
+
+def calculate_psc(data_optcom, optcom_betas):
+    """
+    Calculate percent signal change maps for components against optimally
+    combined data.
+
+    Parameters
+    ----------
+    data_optcom : (S x T) array_like
+        Optimally combined data
+    optcom_betas : (S x C) array_like
+
+    Returns
+    -------
+    PSC : (S x C) array_like
+        Component-wise percent signal change maps.
+    """
+    assert data_optcom.shape[0] == optcom_betas.shape[0]
+    PSC = 100 * optcom_betas / data_optcom.mean(axis=-1, keepdims=True)
+    return PSC
+
+
+def calculate_z_maps(weights, z_max=8):
+    """
+    Calculate z-statistic maps by z-scoring standardized parameter estimate
+    maps and cropping extreme values.
+
+    Parameters
+    ----------
+    weights
+    z_max
+
+    Returns
+    -------
+    Z_maps
+    """
+    Z_maps = stats.zscore(weights, axis=0)
+    extreme_idx = np.abs(Z_maps) > z_max
+    Z_maps[extreme_idx] = z_max * np.sign(Z_maps[extreme_idx])
+    return Z_maps
+
+
+def calculate_f_maps(data_cat, Z_maps, mixing, mask, tes, f_max=500):
+    """
+    Calculate pseudo-F-statistic maps (per component) for TE-dependence
+    and -indepdence models.
```

Comment:
indepENdence

Reply:
@emdupre I can go either way, but my inclination is to merge the disconnected metric modularization code. If we merge now, others can then already start to use those functions and it will make it slightly easier to test things (both general tests & CI testing) and do a clean swap in workflow/tedana.py when we're ready. For the decision tree part, it would also be nice to just have to sync with the master instead of both the master & @tsalo's fork. That said, if @tsalo & I need to tweak a bunch of things to get our parts working together, that might be easier in a fork than needing to have each edit subject to a PR & review.
What are @tsalo's thoughts?